appengine/accessible/workspaceview.component.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Blockly Games: Accessible
+ *
+ * Copyright 2016 Google Inc.
+ * https://github.com/google/blockly-games
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how a Blockly.Workspace is
+ * rendered in AccessibleBlockly. Modified to run music code.
```

Comment:
Well -- on the one hand, I understand the time constraints; on the other hand, this completely breaks the encapsulation of AccessibleBlockly, which is quite serious.

What is the proposed solution? Also, why does the button need to be injected into the block workspace -- can't it live outside it?